packages/graphback-core/src/db/getPrimaryKey.ts

```diff
@@ -0,0 +1,21 @@
+import { GraphQLObjectType, isScalarType } from "graphql";
+import { parseDbAnnotations } from '../annotations/parser';
+import { getBaseType } from '../utils/getBaseType';
+
+export function getPrimaryKey(graphqlType: GraphQLObjectType): string | undefined {
+  const fields = Object.values(graphqlType.getFields());
+
+  let primaryKey: string;
+  for (const field of fields) {
+    const dbConfig: any = parseDbAnnotations(field);
+    const baseType = getBaseType(field.type);
+
+    if (dbConfig.primary) {
+      primaryKey = field.name;
+    } else if (isScalarType(baseType) && baseType.name === 'ID') {
```

Comment:
[Possible bug] It always takes the last and still continues to look for another id. 
We can detect if there are two and pick one with `id` if not fail migration completely as this is clearly bug on the user side.